enhancements/sig-policy/74-order-policy-execution/README.md

```diff
@@ -158,12 +158,14 @@ spec:
     - apiVersion: policy.open-cluster-management.io/v1
       kind: Policy
       name: namespace-foo-setup-policy
+      namespace: policies
       compliance: Compliant
   policy-templates:
     - extraDependencies: //for this policy template only
         - apiVersion: policy.open-cluster-management.io/v1
           kind: ConfigurationPolicy
           name: bar
+          namespace: local-cluster
```

Comment:
Let's not allow or ignore namespaces to be set for policy templates.

Reply:
I'm not sure about allowing `<namespace>.<policy-name>` for `Policy`. My thinking was any dependency on an object defined on the Hub would require a namespace (Policy/PolicySet) and anything only ever created on the managed cluster (ConfigurationPolicy/IAMPolicy/etc) would not have a namespace.

Reply:
This seems like the last tricky thing to figure out. What if the namespace field is (generally) optional in dependencies, and the propagator cleans up / verifies some things before replicating the policy? So:
- it removes the namespace from ConfigurationPolicy, CertificatePolicy, and IAMPolicy dependencies
- for Policy dependencies without a namespace, it tries to extract the namespace from the name, or it gives an error status
- for Policy dependencies with a namespace, the dependency is already in the "right" format
- for PolicySet dependencies, a namespace is required, or it gives an error status. If the namespace is given, it can expand the PolicySet dependency into a list of Policy dependencies (with the namespace on each)
- for other/unknown types, it just keeps them as defined.

Then on the managed cluster, when checking the dependencies, if the namespace is not defined but the resource is namespaced, then it uses the cluster namespace.